parsing/parchmint-parser.js

```diff
@@ -126,6 +125,42 @@ class ParchmintParser {
      */
     connFeatures;
 
+    /**
+     * The max x-span of a placed or routed component or channel.
+     *
+     * @since 1.0.0
+     * @access public
+     *
+     * @type {number}
+     */
+    maxX;
+
+    /**
+     * The max x-span of a placed or routed component or channel.
+     *
+     * @since 1.0.0
+     * @access public
+     *
+     * @type {number}
+     */
+    maxY;
+
+    /**
+     * The padding to add to the max x-/y-spans of the routed component or
+     * channel.
+     *
+     * This is used for the size of the canvas when drawing the SVG, so that
+     * no component/channel is right on the edge of the image.
+     *
+     * Note: This might be a nice setting for the user to adjust on the webpage.
+     *
+     * @since 1.0.0
+     * @access public
+     *
+     * @type {number}
+     */
+    maxPad;
```

Comment:
Is this used when creating the initial rectangle representing the device bounding box or just added around the edge? Either way we definitely need to make it configurable (you want to turn if off when creating a fab file to avoid scaling issues) and we'll probably want to add it to all sides not just past the max.

Reply:
The padding is used to increase the size of the whole SVG image. I didn't know we needed a bounding box, so I'll add that in. How should I determine that size? Should it just be from (0,0) to (maxX, maxY)? We can get rid of the padding all together if you think it's unnecessary.

Reply:
I think the padding is fine as long as its (1) configurable and (2) shows the bounds of the actual device. We can draw a rect from (0,0) to (maxX, maxY) and then add padding along the entire outer edge. However, this means that an offset has to be passed throughout the system to account for the padding to make a correct drawing. I'm guessing this isn't something that was accounted for in the original design, so we may just not want to bother with padding at all for now.

Reply:
You're right I didn't take that into account, so I'll leave out the padding.